api/v1alpha1/rollingupgrade_types.go

```diff
@@ -52,6 +53,85 @@ type RollingUpgradeStatus struct {
 	Conditions              []RollingUpgradeCondition `json:"conditions,omitempty"`
 	LastNodeTerminationTime metav1.Time               `json:"lastTerminationTime,omitempty"`
 	LastNodeDrainTime       metav1.Time               `json:"lastDrainTime,omitempty"`
+
+	Statistics        []*RollingUpgradeStatistics  `json:"statistics,omitempty"`
+	InProcessingNodes map[string]*NodeInProcessing `json:"inProcessingNodes,omitempty"`
+}
+
+// RollingUpgrade Statistics, includes summary(sum/count) from each step
+type RollingUpgradeStatistics struct {
+	StepName      RollingUpgradeStep `json:"stepName,omitempty"`
+	DurationSum   metav1.Duration    `json:"durationSum,omitempty"`
+	DurationCount int32              `json:"durationCount,omitempty"`
+}
+
+// Node In-processing
+type NodeInProcessing struct {
+	NodeName         string             `json:"nodeName,omitempty"`
+	StepName         RollingUpgradeStep `json:"stepName,omitempty"`
+	UpgradeStartTime metav1.Time        `json:"upgradeStartTime,omitempty"`
+	StepStartTime    metav1.Time        `json:"stepStartTime,omitempty"`
+	StepEndTime      metav1.Time        `json:"stepEndTime,omitempty"`
+}
+
+// Add one step duration
+func (s *RollingUpgradeStatus) addStepDuration(asgName string, stepName RollingUpgradeStep, duration time.Duration) {
+	// if step exists, add count and sum, otherwise append
+	for _, s := range s.Statistics {
+		if s.StepName == stepName {
+			s.DurationSum = metav1.Duration{
+				Duration: s.DurationSum.Duration + duration,
+			}
+			s.DurationCount += 1
+			return
+		}
+	}
+	s.Statistics = append(s.Statistics, &RollingUpgradeStatistics{
+		StepName: stepName,
+		DurationSum: metav1.Duration{
+			Duration: duration,
+		},
+		DurationCount: 1,
+	})
+
+	//Add to system level statistics
+	common.AddRollingUpgradeStepDuration(asgName, string(stepName), duration)
+}
+
+// Node turns onto step
+func (s *RollingUpgradeStatus) NodeTurnsOntoStep(asgName string, nodeName string, stepName RollingUpgradeStep) {
```

Comment:
@shaoxt just a suggestion. I think NodeStep(...) will be a better naming than NodeTurnsOntoStep(...)